ui/server/src/main/scala/pl/touk/nussknacker/ui/process/migrate/RemoteEnvironment.scala

```diff
@@ -182,21 +178,27 @@ trait StandardRemoteEnvironment extends Argonaut62Support with RemoteEnvironment
   override def testMigration(implicit ec: ExecutionContext): Future[Either[EspError, List[TestMigrationResult]]] = {
     (for {
       basicProcesses <- EitherT(invokeJson[List[BasicProcess]](HttpMethods.GET, List("processes")))
-      processes      <- fetchGroupByGroup(basicProcesses, basicProcess => EitherT(invokeJson[ValidatedProcessDetails](HttpMethods.GET, List("processes", basicProcess.name))))
+      processes      <- fetchGroupByGroup(basicProcesses)
       subprocesses   <- EitherT(invokeJson[List[ValidatedProcessDetails]](HttpMethods.GET, List("subProcessesDetails")))
     } yield testModelMigrations.testMigrations(processes, subprocesses)).value
   }
 
-  private def fetchGroupByGroup[T](basicProcesses: List[BasicProcess],
-                                   getOne: BasicProcess => EitherT[Future, EspError, ValidatedProcessDetails])
+  private def fetchGroupByGroup[T](basicProcesses: List[BasicProcess])
                                   (implicit ec: ExecutionContext): EitherT[Future, EspError, List[ValidatedProcessDetails]] = {
-    EitherT(Future.successful(Await.result(
-      Future.sequence(
-        basicProcesses.grouped(config.migrationBatchSize)
-          .flatMap(_.map(getOne(_).value))
-          .toList
-      ).map(_.sequence),
-      config.migrationTimeout
-    )))
+    def getMany(processes: List[BasicProcess]) = {
+      invokeJson[List[ValidatedProcessDetails]](
+        HttpMethods.POST,
+        "processesDetails" :: Nil,
+        requestEntity = HttpEntity(ContentTypes.`application/json`, processes.map(_.name).asJson.nospaces))
+    }
+
+    EitherT(Future.successful(
+      basicProcesses.grouped(config.batchSize)
+        .foldLeft(List.empty[ValidatedProcessDetails].asRight[EspError]) { case (acc, batch) => for {
+            current <- acc
+            fetched <- Await.result(getMany(batch), config.batchTimeout)
```

Comment:
can we do it without await?
If we don't want to sequence because of thread pool limitations, maybe we can do flatmap instead?
Then timeout wouldn't be needed?

Reply:
good point, fixed that